core/src/main/scala/chisel3/Trie.scala

```diff
@@ -0,0 +1,135 @@
+package chisel3
+
+
+import scala.annotation.tailrec
+import scala.collection.mutable
+
+/** A prefix tree. Prefixes represented by a list of `Token` values. For each
+  * valid path the tree will contain a value of type `Value`.
+  *
+  * To use this, do not extend this trait (it is sealed). Instead, create an empty trie with
```

Comment:
can you provide some wikipedia reference or something to what a Trie is?

Reply:
Yup!